src/main/java/org/java_lcw/jil/AWTImage.java

```diff
@@ -0,0 +1,485 @@
+    case BufferedImage.TYPE_4BYTE_ABGR: {
+      return new AWTImage(Image.MODE_RGBA, bi.getWidth(), bi.getHeight(), bi);
+    }
+
+    case BufferedImage.TYPE_4BYTE_ABGR_PRE:
+    case BufferedImage.TYPE_INT_ARGB:
+    case BufferedImage.TYPE_INT_ARGB_PRE:
+    default: {
+      BufferedImage nbi = new BufferedImage(bi.getWidth(), bi.getHeight(), BufferedImage.TYPE_4BYTE_ABGR);
+      nbi.getGraphics().drawImage(bi, 0, 0, null);
+      return new AWTImage(Image.MODE_RGBA, nbi.getWidth(), nbi.getHeight(), nbi);
+    }
+
+    }
+  }
+
+  public static AWTImage open(String filename) throws ImageException, IOException {
+    try {
+      return open(filename, Utils.getImageType(filename));
+    } catch(ImageException e) {
+      for(ImageType t: ImageType.values()) {
+        try {
+          return open(filename, t);
+        } catch(ImageException e1) {
+        }
+      }
+      throw new ImageException("Could not figure out image type!");
+    }
+  }
+
+  public static AWTImage open(String filename, ImageType type) throws IOException, ImageException {
+    switch(type) {
+    case TIFF:
+      return AWTImage.fromJavaImage(TiffFile.open(filename));
+    case PNG:
+    case JPEG:
+      return AWTImage.fromBufferedImage(ImageIO.read(new File(filename)));
+    default:
+      throw new ImageException("Could not determen filetype");
+    }
+  }
+
+  public BufferedImage getBufferedImage() {
+    return bi;
+  }
+
+  @Override
+  public void save(File file) throws IOException, ImageException {
+    ImageType t = Utils.getImageType(file);
+    save(file, t);
+  }
+
+  @Override
+  public void save(String file) throws IOException, ImageException {
+    ImageType t = Utils.getImageType(file);
+    save(file, t);
+  }
+
+  @Override
+  public void save(File file, ImageType type) throws IOException, ImageException {
+    save(file.getAbsolutePath(), type);
+  }
+
+  @Override
+  public void save(String filename, ImageType type) throws IOException, ImageException {
+    switch(type) {
+    case JPEG: {
+      JpegFile.save(filename, this);
+    }break;
+    case PNG: {
+      PngFile.save(filename, this);
+    } break;
+    case TIFF: {
+      TiffFile.save(filename, this);
+    } break;
+    default:
+      break;
+    }
+  }
+
+
+  @Override
+  public JavaImage toJavaImage() {
+    return JavaImage.fromBufferedImage(bi);
+  }
+
+  @Override
+  public AWTImage changeMode(byte MODE) {
+    BufferedImage nbi;
+    if(MODE == Image.MODE_L) {
+      nbi = new BufferedImage(width, height, BufferedImage.TYPE_BYTE_GRAY);
+    } else if(MODE == Image.MODE_RGB) {
+      nbi = new BufferedImage(width, height, BufferedImage.TYPE_3BYTE_BGR);
+    } else {
+      nbi = new BufferedImage(width, height, BufferedImage.TYPE_4BYTE_ABGR);
+    }
+    nbi.getGraphics().drawImage(bi, 0, 0, null);
+    return new AWTImage(nbi, MODE);
+  }
+
+  @Override
+  public AWTImage resizeWithBorders(int bWidth, int bHeight, Color borderColor,
+      ScaleType st) {
+    return null;
```

Comment:
Technically its not unsupported, I just didnt implement it... I will add an exception for now